[blog]: How to add blog post to OSCSA blog (#46)
Open-Science-Community-Saudi-Arabia/OSCSA-en-blog: posts/How-to-add-blog-post/index.qmd

---
title: "How to add a blog post using Quarto"
author: 
    name: "Richie Moluno, Goodnews Sandygudie"
date: "2022-12-08"
categories: 
    - "blog"
    - "quarto"
    - "Outreachy"
description: "This is a tutorial on how to add a blog post to the OSCA blog using Quarto."
licence: "CC BY"
type: "post"
toc-title: "Table of Contents"
toc-location: right
toc-depth: 3
toc: true
image: "./quarto.png"
citation: 
    url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
site-url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
---

#

![](./quarto.png){fig-align="center" width="500" fig-caption="Quarto logo"}

#

## What is Quarto?
Quarto is a new open source project that aims to make it easier for people to create and publish blog posts. It is a static site generator that uses the Markdown syntax to create blog posts. Quarto is written in the R programming language and is available for download on GitHub.

## Get started
### Prerequisites
To get started with contributing with Quarto, you will need;

- A GitHub account.
- A basic understanding of Markdown is recommended but not necessary.

> ### Install Quarto

Reviewer: I don't think they need to install Quarto - so maybe this paragraph can be moved to the side using `{.aside}`

Author: Oh, alright thanks

Author: done

- Next, you will need to create a pull request to merge your changes to the main branch. You can do this by navigating to the GitHub repository and clicking on the `New pull request` button.

![](./new_PR_new.png){fig-align="center" width="80%"}

- The branch for the base repository should be set to `main` and the branch for the head repository should be set to the branch that you created in the previous step. Click the `Create pull request` button to create a new pull request.

![](./new_PR1.png){fig-align="center" width="80%"}

- You can then add a title and description for your pull request. Click the `Create pull request` button to create a new pull request.

![](./new_PR2.png){fig-align="center" width="80%"}

The title of your pull request should be a short description of the changes that you made.
it should follow the format `[blog]: Title of your blog`

Automatically, the PR template will be added to the description of your pull request. You can add more information about your pull request in the description.

#

##### Step 8 - Merge your changes
Once you have created a pull request, you can wait for your changes to be reviewed and merged to the main branch.

> #

Reviewer: Can you show here a sample of a blog post with YAML (very small sample, what the end product looks like?)

Author: Alright

Author: by sample do you mean the rendered preview?

Author: I think i get it now, you meant a YAML code preview right

### Preview your blog post
Since we are working directly from the Github account and Quarto is a command line tool, you can not preview the HTML rendered version of your blogpost.
To see the preview:

> 

Reviewer: Just show how they can preview it from GitHub without the command line!

Author: I didn't know it is possible to preview without using the terminal

Author: I'll check the documentation on how to do it

Author: I used the netlify preview link instead, is this okay?

Reviewer: Yes, that's what I meant, thank you!

---
title: "How to add a blog post using Quarto"
author: 
    name: "Richie Moluno, Goodnews Sandygudie"
date: "2022-12-08"
categories: 
    - "blog"
    - "quarto"
    - "Outreachy"
description: "This is a tutorial on how to add a blog post to the OSCA blog using Quarto."
licence: "CC BY"
type: "post"
toc-title: "Table of Contents"
toc-location: right

> toc-depth: 3

Reviewer: ```suggestion
toc-depth: 5
```

toc: true
image: "./quarto.png"
citation: 
    url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
site-url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
---

#

![](./quarto.png){fig-align="center" width="500" fig-caption="Quarto logo"}

#

## What is Quarto?
Quarto is a new open source project that aims to make it easier for people to create and publish blog posts. It is a static site generator that uses the Markdown syntax to create blog posts. Quarto is written in the R programming language and is available for download on GitHub.

## Get started
### Prerequisites
To get started with contributing with Quarto, you will need;

> - A GitHub account.

Reviewer: Add a link to the GitHub
```suggestion
- A [GitHub](https://github.com/) account.
```

Author: done

> - A basic understanding of Markdown is recommended but not necessary.

Reviewer: The same with Markdown, need to link it to any cheatsheet for Markdown.
```suggestion
- A basic understanding of Markdown is recommended but not necessary.
```

Author: done

::: {.aside}
### Install Quarto
To install quarto on your computer, you can download the latest version of R from the R Project website. 

To Download the latest version of quarto visit the [Quarto](https://quarto.org/docs/get-started/) website.
:::
#

### Setup Local environment
To start making contributions to the OSCA blog, you will need to setup a personal environment on your personal Github account. This will allow you to make changes to the blog without affecting the live version of the blog.

#

##### Step 1 - Fork the repository

> To get started, you will need to fork the OSCA blog repository from GitHub.

Reviewer: Add links
```suggestion
To get started, you will need to fork the [OSCA blog repository](https://github.com/Open-Science-Community-Saudi-Arabia/OSCSA-en-blog/) from GitHub.
```

Author: @sandygudie had fixed this

## What is Quarto?
![](./quarto.png){fig-align="center" width="100%"}   <!-- Image file -->

Mollit eiusmod culpa in ad esse nulla est do consectetur. Eiusmod et Lorem sint veniam nisi minim ad velit sit culpa. Occaecat culpa sint cupidatat minim elit commodo labore mollit est ipsum. Anim officia est tempor esse voluptate occaecat nulla velit elit officia elit consectetur enim. Aliqua est reprehenderit culpa labore magna ut deserunt. Proident ex ad quis exercitation dolore minim cupidatat ad pariatur. Est minim irure enim non incididunt velit laborum est adipisicing enim eiusmod quis.

```

To see a preview of your blog post, you can visit the netlify preview link that was generated when you created the pull request. 

![](./preview_link.png){fig-align="center" width="80%"}

#

### Conclusion

> In this tutorial, we learned how to create a blog post using Quarto. We started by installing Quarto and setting up our local environment. We then created a new blog post using the Quarto syntax and previewed it using the Quarto serve command. Finally, we submitted a pull request for the new changes.

Reviewer: ```suggestion
In this tutorial, we learned how to create a blog post using Quarto. We started by installing Quarto and setting up our local environment. We then created a new blog post using the Quarto syntax. Finally, we submitted a pull request for the new changes and previewed it using the Netlify Preview.
```

---

> title: "How to add a blog post using Quarto"

Reviewer: ```suggestion
title: "How to create a blog post in the Open Science Community Saudi Arabia using Quarto?"
```

Author: alright, noted, i'll make the necessary changes
